forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/services/openbanking/event/EventResponseUtilService.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.common.services.openbanking.event;
+
+import com.forgerock.openbanking.common.conf.discovery.ResourceLinkService;
+import com.forgerock.openbanking.common.model.openbanking.v3_0.event.FRCallbackUrl1;
+import com.forgerock.openbanking.common.model.version.OBVersion;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.org.openbanking.datamodel.account.Links;
+import uk.org.openbanking.datamodel.account.Meta;
+import uk.org.openbanking.datamodel.event.*;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Util Service class to handler the Event Notification API responses thru all versions<br />
+ */
+@Service
+@Slf4j
+public class EventResponseUtilService {
+
+    private final ResourceLinkService resourceLinkService;
+    private OBVersion version;
+    private boolean shouldHaveMetaSection;
+
+    public EventResponseUtilService(ResourceLinkService resourceLinkService) {
+        this.resourceLinkService = resourceLinkService;
+        this.shouldHaveMetaSection = true;
+        this.version = null;
+    }
+
+    public OBVersion getVersion() {
+        return version;
+    }
+
+    public EventResponseUtilService setVersion(OBVersion version) {
+        this.version = version;
+        return this;
+    }
+
+    public boolean isShouldHaveMetaSection() {
+        return shouldHaveMetaSection;
+    }
+
+    public EventResponseUtilService setShouldHaveMetaSection(boolean shouldHaveMetaSection) {
```

Comment:
It looks like this is basically true if the version is later than 3.0 - is that right?

There's a lot of logic being forced into the clients of this class and the associated code around it here. Why don't you just check if the version is > 3.0 within the `packageResponse()` methods in this class and do away with this property?